File: scripts/send_notification.py
"""
Simple push notification sender.

Usage:
    python scripts/send_notification.py <push_token> [title] [body]

Examples:
    python scripts/send_notification.py "ExponentPushToken[abc123]" "Hello" "World"
    python scripts/send_notification.py "<fcm-token>" "Workout done" "Great job!"
"""

import sys
import os
import json
import urllib.request

def send_expo(token: str, title: str, body: str) -> dict:
    payload = json.dumps({
        "to": token,
        "title": title,
        "body": body,
        "sound": "default",
    }).encode()

    req = urllib.request.Request(
        "https://exp.host/--/api/v2/push/send",
        data=payload,
        headers={"Content-Type": "application/json"},
        method="POST",
    )
    with urllib.request.urlopen(req) as res:
        return json.loads(res.read())


def send_fcm(token: str, title: str, body: str) -> dict:
    server_key = os.environ.get("FCM_SERVER_KEY")
    if not server_key:
        print("Error: set FCM_SERVER_KEY env variable for FCM tokens.", file=sys.stderr)
        sys.exit(1)

    payload = json.dumps({
        "to": token,
        "notification": {"title": title, "body": body, "sound": "default"},
    }).encode()

    req = urllib.request.Request(
        "https://fcm.googleapis.com/fcm/send",
        data=payload,
        headers={
            "Content-Type": "application/json",
            "Authorization": f"key={server_key}",
        },
        method="POST",
    )
    with urllib.request.urlopen(req) as res:
        return json.loads(res.read())


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python send_notification.py <push_token> [title] [body]")
        sys.exit(1)

    push_token = sys.argv[1]
    title = sys.argv[2] if len(sys.argv) > 2 else "Momentum"
    body = sys.argv[3] if len(sys.argv) > 3 else "You have a new notification"

    is_expo = push_token.startswith("ExponentPushToken")
    print(f"Sending via {'Expo' if is_expo else 'FCM'}...")

    result = send_expo(push_token, title, body) if is_expo else send_fcm(push_token, title, body)
    print("Response:", json.dumps(result, indent=2))

File: scripts/send-notification.js
/**
 * Simple push notification sender.
 *
 * Usage:
 *   node scripts/send-notification.js <pushToken> [title] [body]
 *
 * Examples:
 *   node scripts/send-notification.js ExponentPushToken[xxxxxx] "Hello" "World"
 *   node scripts/send-notification.js <fcm-token> "Workout complete" "Great job!"
 */

const [, , pushToken, title = "Momentum", body = "You have a new notification"] =
  process.argv;

if (!pushToken) {
  console.error("Usage: node send-notification.js <pushToken> [title] [body]");
  process.exit(1);
}

// --- Expo Push (works if token starts with ExponentPushToken) ---
async function sendExpo() {
  const res = await fetch("https://exp.host/--/api/v2/push/send", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      to: pushToken,
      title,
      body,
      sound: "default",
    }),
  });
  return res.json();
}

// --- FCM Legacy HTTP (requires FCM_SERVER_KEY env var) ---
async function sendFCM() {
  const serverKey = process.env.FCM_SERVER_KEY;
  if (!serverKey) {
    console.error("Set FCM_SERVER_KEY env variable for FCM tokens.");
    process.exit(1);
  }
  const res = await fetch("https://fcm.googleapis.com/fcm/send", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `key=${serverKey}`,
    },
    body: JSON.stringify({
      to: pushToken,
      notification: { title, body, sound: "default" },
    }),
  });
  return res.json();
}

(async () => {
  const isExpo = pushToken.startsWith("ExponentPushToken");
  console.log(`Sending via ${isExpo ? "Expo" : "FCM"}...`);

  const result = isExpo ? await sendExpo() : await sendFCM();
  console.log("Response:", JSON.stringify(result, null, 2));
})();
